src/dhcp/server.c

```diff
@@ -456,7 +456,7 @@ static void init_packet(GDHCPServer *dhcp_server, struct dhcp_packet *packet,
 	packet->gateway_nip = client_packet->gateway_nip;
 	packet->ciaddr = client_packet->ciaddr;
 	dhcp_add_option_uint32(packet, DHCP_SERVER_ID,
-						dhcp_server->server_nip);
+						get_be32(&dhcp_server->server_nip));
```

Comment:
Cool. I was reading gitter.im and found all your responses. There's a new feature on gitter (conversations) that would help to follow our conversation. Did you test if this works between two miraclecast?